android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -66,6 +64,63 @@ public Screen getScreen() {
     return mScreenView;
   }
 
+  public void onContainerUpdate() {
+    ScreenStackHeaderConfig config = findHeaderConfig();
+    boolean configInChild = hasChildScreenWithConfig(getScreen());
+    if (!configInChild && config != null && config.getScreenFragment().getActivity() != null) {
+      // if there is no child with config, we look for a parent with config to set the orientation
+      config.getScreenFragment().getActivity().setRequestedOrientation(config.getScreenOrientation());
+    }
+  }
+
+  private ScreenStackHeaderConfig findHeaderConfig() {
+    ViewParent parent = getScreen().getContainer();
+    while (parent != null) {
+      if (parent instanceof Screen) {
+        ScreenStackHeaderConfig headerConfig = ((Screen) parent).getHeaderConfig();
+        if (headerConfig != null) {
+          return headerConfig;
+        }
+      }
+      parent = parent.getParent();
+    }
+    return null;
+  }
+
+  protected boolean hasChildScreenWithConfig(Screen screen) {
+    if (screen == null) {
+      return false;
+    }
+    for (ScreenContainer sc : screen.getFragment().getChildScreenContainers()) {
+      if (sc instanceof ScreenStack) {
+        // we check only the top screen of stack for header config
+        Screen topScreen = ((ScreenStack) sc).getTopScreen();
+        ScreenStackHeaderConfig headerConfig = topScreen != null ? topScreen.getHeaderConfig(): null;
+        if (headerConfig != null) {
+          return true;
+        }
+        if (hasChildScreenWithConfig(topScreen)) {
+          return true;
+        }
+      } else {
+        // in ScreenContainer we check only the screen that is on top
+        for (int i = 0; i < sc.getScreenCount(); i++) {
+          Screen childScreen = sc.getScreenAt(i);
+          if (childScreen.getActivityState() == Screen.ActivityState.ON_TOP) {
+            if (hasChildScreenWithConfig(childScreen)) {
```

Comment:
Those ifs can be merged

Reply:
It may result in worse code readability, but I have no strong opinion on that.

Reply:
I think writing this as a conditions' conjunction will be more readable than introducing another indent.